source/_docs/configuration/templating.markdown

```diff
@@ -195,7 +195,19 @@ The same thing can also be expressed as a filter:
 - Filter `timestamp_utc` converts a UNIX timestamp to its string representation representation as date/time in UTC timezone.
 - Filter `timestamp_custom(format_string, local_time=True)` converts an UNIX timestamp to its string representation based on a custom format, the use of a local timezone is default. Supports the standard [Python time formatting options](https://docs.python.org/3/library/time.html#time.strftime).  
 
-Note: [UNIX timestamp](https://en.wikipedia.org/wiki/Unix_time) is the number of seconds that have elapsed since 00:00:00 UTC on 1 January 1970. Therefore, if used as a function's argument, it can be substituted with a numeric value (`int` or `float`):  
+<div class='note'>
+[UNIX timestamp](https://en.wikipedia.org/wiki/Unix_time) is the number of seconds that have elapsed since 00:00:00 UTC on 1 January 1970. Therefore, if used as a function's argument, it can be substituted with a numeric value (`int` or `float`):  
+</div>
+
+<div class='note warning'>
+If your template is returning a timestamp that should be displayed in the frontend (e.g., as a sensor entity with `device_class: timestamp`), you have to ensure that it is the ISO 8601 format (meaning it has the "T" separator between the date and time portion). Otherwise, frontend rendering on macOS and iOS devices will show an error. For Home Assistant internal timestamp fields such as an entity's `last_change` and `last_update` this is done out of the box. However, the following value template would result in the frontend error:
+
+`{{ states.sun.sun.last_updated }}` => `2021-01-04 12:27:08.036538+00:00` (missing "T" separator)
```

Comment:
We should fix this in the frontend... to allow for more slack on this part... 

As in, that would be the more user-friendly approach.

Reply:
For non-templates the backend takes care of that. For the internal "last_changed" and "last_updated" there are conversions in the core (core.py => as_dict()) and for integrations we mandate that they use ISO 8601. If the template logic also would by default get isoformat() from the core for the evaluation, then the frontend would not have to attempt to fix any timestamps.

Reply:
The template cannot fix that, as the output is user output/input. And we don't know its intention. Therefore the suggestion to maybe gracefully handle it in the frontend maybe? Dunno...

Reply:
Never looked at the template coding, but my idea would have been: 

IF (last_changed || last_updated) is requested by template engine:
THEN provide as isoformat()

But that would leave those scenario where as user is creating their own fresh timestamps in the template not based on any HA states / attributes.

I can discuss with Bram if he is okay with the frontend doing a conversion (`replace(" ", "T")`) every time (perhaps we can filter to do that only for MacOS / iOS somehow)?

=> For now still merge this note, as currently the user has to handle it?

Reply:
We can do a `replace(" ", "T")` in the date function.